src/maker/config.rs

```diff
@@ -32,20 +34,94 @@ pub struct MakerConfig {
+                .get("absolute_fee_sats")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            amount_relative_fee_ppb: maker_config_section
+                .get("amount_relative_fee_ppb")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            time_relative_fee_ppb: maker_config_section
+                .get("time_relative_fee_ppb")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            required_confirms: maker_config_section
+                .get("required_confirms")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            min_contract_reaction_time: maker_config_section
+                .get("min_contract_reaction_time")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            min_size: maker_config_section
+                .get("min_size")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            idle_connection_timeout: maker_config_section
+                .get("idle_connection_timeout")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            watchtower_ping_interval_secs: maker_config_section
+                .get("watchtower_ping_interval_secs")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            heart_beat_interval_secs: maker_config_section
+                .get("heart_beat_interval_secs")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            onion_addrs,
+            rpc_ping_interval_secs: maker_config_section
+                .get("rpc_ping_interval_secs")
+                .unwrap()
+                .parse()
+                .unwrap(),
+            directory_servers_refresh_interval_secs: maker_config_section
+                .get("directory_servers_refresh_interval_secs")
+                .unwrap()
+                .parse()
+                .unwrap(),
+        }
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use super::MakerConfig;
+
+    #[test]
+    fn test_parsing_toml_maker() {
+        let dummy_maker = MakerConfig {
+            port: 3000,
             heart_beat_interval_secs: 3,
             rpc_ping_interval_secs: 60,
             watchtower_ping_interval_secs: 300,
             directory_servers_refresh_interval_secs: 60 * 60 * 12, //12 Hours
             idle_connection_timeout: 300,
-            onion_addrs,
+            onion_addrs: "example.onion".to_string(),
             absolute_fee_sats: 1000,
             amount_relative_fee_ppb: 10_000_000,
             time_relative_fee_ppb: 100_000,
             required_confirms: 1,
             min_contract_reaction_time: 48,
             min_size: 10_000,
-        }
+        };
+
+        let parsed_maker = MakerConfig::init(3000, "example.onion".to_string());
+
+        assert_eq!(dummy_maker, parsed_maker);
```

Comment:
This test only works as long as there is a fully filled conf files. What happens when the file is missing? Or the user deleted it by mistake, or some lines are missing in the conf files.

I.e, default value handling needs to be added.

If the file is missing, the software should create the file and write the default values into it. Same thing if a config line is missing.  

And all this should be tested.